zebra/zebra_nb_config.c

```diff
@@ -1279,17 +1289,21 @@ int lib_vrf_zebra_l3vni_id_destroy(struct nb_cb_destroy_args *args)
 	case NB_EV_PREPARE:
 	case NB_EV_ABORT:
 	case NB_EV_VALIDATE:
-		break;
+		return NB_OK;
 	case NB_EV_APPLY:
 		vrf = nb_running_get_entry(args->dnode, NULL, true);
-		zvrf = zebra_vrf_lookup_by_id(vrf->vrf_id);
+		zvrf = zebra_vrf_lookup_by_name(vrf->name);
 		vni = yang_dnode_get_uint32(args->dnode, NULL);
 
 		if (!zl3vni_lookup(vni))
 			return NB_OK;
 
-		if (zvrf->l3vni != vni)
+		if (zvrf->l3vni != vni) {
+			snprintf(args->errmsg, args->errmsg_len,
```

Comment:
snprintf?

Reply:
Configuration rejection error(s) need to relayed to user via errmsg which is northbound framework buffer.